gitutil.py

```diff
@@ -130,14 +130,24 @@ def branch_head(self, branch_name):
         # XXX added for backward compatibility - remove
         return self.fetch_head(ref=branch_name)
 
-
-def clone_repository(
-        to_path: str,
-        github_cfg,
-        github_repo_path: str
-    ):
-        url = url_with_credentials(github_cfg, github_repo_path)
-        git.Git(to_path).clone('--quiet', url)
+    @staticmethod
+    def clone_into_directory(
+            directory_path: str,
+            github_cfg,
+            github_repo_path: str,
+            branch_name: str = 'Master'
```

Comment:
I think, this should be written with a lower-cased 'm'

also: how about naming this 'initial_ref' (and setting the default value to refs/heads/master)?
also, consider using None as default value.
there is why: this way, we can discriminate between cases where the caller explicitly gave a ref and those where they did not.
in the first case, also add the '--single-branch' CLI to git-clone. This way, we do not fetch objects not reachable from the given initial ref.